helper.php

```diff
@@ -125,6 +125,42 @@ public function getTokenEndpoint($service) {
         $service = strtolower($service);
         return $this->getConf($service.'-tokenurl');
     }
+
+    /**
+     * @param bool $string if true returns a nice string for output, otherwise returns array of strings
+     *
+     * @return array|string
+     */
+    public function getValidDomains($string = false) {
+        $validDomains = explode(',', trim($this->getConf('mailRestriction'), ','));
+        if ($string) {
+            $domainListing = $validDomains[0];
+            array_shift($validDomains);
+            while (count($validDomains) > 0) {
```

Comment:
I would do this joining to a string outside the function. Let this function always return an array.